# ScaleBio Universal ATAC: Citations

## Workflow creation 
- [nf-core](https://pubmed.ncbi.nlm.nih.gov/32055031/)
    > Ewels PA, Peltzer A, Fillinger S, Patel H, Alneberg J, Wilm A, Garcia MU, Di Tommaso P, Nahnsen S. The nf-core framework for community-curated bioinformatics pipelines. Nat Biotechnol. 2020 Mar;38(3):276-278. doi: 10.1038/s41587-020-0439-x. PubMed PMID: 32055031.

- [Nextflow](https://pubmed.ncbi.nlm.nih.gov/28398311/)
    > Di Tommaso P, Chatzou M, Floden EW, Barja PP, Palumbo E, Notredame C. Nextflow enables reproducible computational workflows. Nat Biotechnol. 2017 Apr 11;35(4):316-319. doi: 10.1038/nbt.3820. PubMed PMID: 28398311.

## Pipeline tools 

- [BEDTools](https://pubmed.ncbi.nlm.nih.gov/20110278/)
    > Quinlan AR, Hall IM. BEDTools: a flexible suite of utilities for comparing genomic features. Bioinformatics. 2010 Mar 15;26(6):841-2. doi: 10.1093/bioinformatics/btq033. Epub 2010 Jan 28. PubMed PMID: 20110278; PubMed Central PMCID: PMC2832824.

- [Bowtie 2](https://github.com/BenLangmead/bowtie2)
    > Langmead B, Salzberg S. Fast gapped-read alignment with Bowtie 2. Nature Methods. 2012, 9:357-359.

- [Cutadapt](https://cutadapt.readthedocs.io/en/stable/)
    > Martin M. Cutadapt Removes Adapter Sequences From High-Throughput Sequencing Reads. EMBnet Journal. 2011 

- [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
    > Bioinformatics B. FastQC: a quality control tool for high throughput sequence data. Cambridge, UK: Babraham Institute. 2011.

- [MACS3](https://github.com/macs3-project/MACS)
    > Zhang, Yong, et al. "Model-based analysis of ChIP-Seq (MACS)." Genome biology. 2008, 9.9:1-9.

- [Samtools](https://github.com/samtools/samtools)
    > Li H, Handsaker B, Wysoker A el al., The Sequence Alignment/Map format and SAMtools. Bioinformatics. 2009 Aug 15;25(16):2078-9. doi: 10.1093/bioinformatics/btp352. Epub 2009 Jun 8. PMID: 19505943; PMCID: PMC2723002.

## Software packaging/containerisation tools
- [Anaconda](https://anaconda.com)
    > Anaconda Software Distribution. Computer software. Vers. 2-2.4.0. Anaconda, Nov. 2016. Web.

- [Bioconda](https://pubmed.ncbi.nlm.nih.gov/29967506/)
    > Grüning B, Dale R, Sjödin A, Chapman BA, Rowe J, Tomkins-Tinch CH, Valieris R, Köster J; Bioconda Team. Bioconda: sustainable and comprehensive software distribution for the life sciences. Nat Methods. 2018 Jul;15(7):475-476. doi: 10.1038/s41592-018-0046-7. PubMed PMID: 29967506.

- [Docker](https://dl.acm.org/doi/10.5555/2600239.2600241)

- [Singularity](https://pubmed.ncbi.nlm.nih.gov/28494014/)
    > Kurtzer GM, Sochat V, Bauer MW. (2017). Singularity: Scientific containers for mobility of compute. PLoS One. 2017 May 11;12(5):e0177459. eCollection . PubMed PMID: 28494014; PubMed Central PMCID: PMC5426675.

## Python Packages 
- [Datapane](https://github.com/datapane/datapane)

- [Matplotlib](https://matplotlib.org/)
    > J. D. Hunter, (2007). "Matplotlib: A 2D Graphics Environment", Computing in Science & Engineering, vol. 9, no. 3, pp. 90-95. 

- [Numpy](https://numpy.org/)
    > Harris, C. R., et al., (2020). Array programming with NumPy. Nature, 585, 357–362. 

- [Pandas](https://pandas.pydata.org/)
    > McKinney, W., & others. (2010). Data structures for statistical computing in python. In Proceedings of the 9th Python in Science Conference (Vol. 445, pp. 51–56).

- [Plotly](https://plot.ly) 
    > Plotly Technologies Inc. (2015). Collaborative data science Publisher: Plotly Technologies Inc.: Montréal, QC. 

- [Pybedtools](https://daler.github.io/pybedtools/)
    > Dale RK, Pedersen BS, and Quinlan AR. (2011). Pybedtools: a flexible Python library for manipulating genomic datasets and annotations. Bioinformatics 27(24):3423-3424.

- [Seaborn](https://github.com/mwaskom/seaborn)
    > Waskom, M. L., (2021). seaborn: statistical data visualization. Journal of Open Source Software, 6(60), 3021


## R Packages 
- [AnnotationHub](https://github.com/Bioconductor/AnnotationHub)
    > Martin Morgan and Lori Shepherd (2022). AnnotationHub: Client to access AnnotationHub resources. R package version 3.2.2.

- [ArchR](https://www.archrproject.com/index.html)
    > Jeffrey Granja and Ryan Corces (2020). ArchR: Analyzing single-cell regulatory chromatin in R.. R package version 1.0.1.

- [argparse](https://CRAN.R-project.org/package=argparse)
    > Trevor L Davis (2021). argparse: Command Line Optional and Positional Argument Parser. R package version 2.1.3.

- [BiocManager](https://CRAN.R-project.org/package=BiocManager)
    > Martin Morgan (2021). BiocManager: Access the Bioconductor Project Package Repository. R package version 1.30.16.

- [BSGenome](https://bioconductor.org/packages/BSgenome)
    > Hervé Pagès (2021). BSgenome: Software infrastructure for efficient representation of full genomes and their SNPs. R package version 1.62.0.

- ['dplyr'](https://CRAN.R-project.org/package=dplyr)
    > Hadley Wickham, Romain François, Lionel Henry and Kirill Müller (2022). dplyr: A Grammar of Data Manipulation. R package version

- ['GenomicRanges'](http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118)
    > Lawrence M, Huber W, Pages H, Aboyoun P, Carlson M, et al. (2013) Software for Computing and Annotating Genomic Ranges. PLoS Comput Biol 9(8)

- [ggplot2](https://cran.r-project.org/web/packages/ggplot2/index.html)
    > H. Wickham. (2016) ggplot2: Elegant Graphics for Data Analysis. Springer-Verlag New York

- [glue](https://CRAN.R-project.org/package=glue)
    > Jim Hester and Jennifer Bryan (2022). glue: Interpreted String Literals. R package version 1.6.2.

- [Seurat](https://github.com/satijalab/seurat)
    > Hao and Hao et al. Integrated analysis of multimodal single-cell data. Cell (2021) [Seurat V4]

- [yaml](https://CRAN.R-project.org/package=yaml)
    > Shawn P Garbett, Jeremy Stephens et al., (2022). yaml: Methods to Convert R Data to YAML and Back. R package version 2.3.5.

